src/Cassandra/Data/Linq/CqlExpressionVisitor.cs

```diff
@@ -72,15 +86,19 @@ internal class CqlExpressionVisitor : ExpressionVisitor
         private int _limit;
         private readonly List<Tuple<PocoColumn, object, ExpressionType>> _projections = new List<Tuple<PocoColumn, object, ExpressionType>>();
         private readonly List<Tuple<string, bool>> _orderBy = new List<Tuple<string, bool>>();
-        private readonly List<string> _selectFields = new List<string>();
+        private readonly List<string> _selectFields = new List<string>(DefaultClauseParameterCapacity);
```

Comment:
I agree it has to be a guess. So, it comes down to: allocate more memory that would not be used or resizing the array if the select has more than 8 columns. I am 👍 with the fixed value.